### PR TITLE
Verify overlay cache is accurately and fully tested

### PR DESCRIPTION
**Pull request by wthorp**
fixed test using valid port ranges, go routines, and node ids

### CODE REVIEW

**Comment by wthorp** [comment]
The test that was previously skip()'ed now passes.  That's really all I've done here.

File: pkg/overlay/cache_test.go [hunk 1]
```diff
@@ -80,8 +80,10 @@ func bootstrapTestNetwork(t *testing.T, ip, port string) ([]dht.DHT, pb.Node) {
 	assert.NoError(t, err)
 	bootNode := rt.Local()
 
-	err = boot.ListenAndServe()
-	assert.NoError(t, err)
+	go func() {
```
Comment (egonelbre) [inline pkg/overlay/cache_test.go]: Btw I created a `testcontext` package that helps with these sorts of tests.

```
ctx := testcontext.New(t)
defer ctx.Shutdown()
...
ctx.Go(boot.ListenAndServe)
```

Comment (wthorp) [reply pkg/overlay/cache_test.go]: I'm not sure why, but using textcontext causes my tests to timeout.  Maybe a deadlock w/ the waitgroup?

ctx := testcontext.New(t)
defer ctx.Cleanup()
ctx.Go(boot.ListenAndServe)
^^ this fails

File: pkg/overlay/cache.go [hunk 2]
```diff
@@ -115,17 +109,20 @@ func (o *Cache) Put(nodeID string, value pb.Node) error {
...
 		return err
 	}
-
 	return o.DB.Put(node.IDFromString(nodeID).Bytes(), data)
 }
 
 // Bootstrap walks the initialized network and populates the cache
 func (o *Cache) Bootstrap(ctx context.Context) error {
-	nodes, err := o.DHT.GetNodes(ctx, "", 1280)
+	r, err := randomID()
```
Comment (jenlij) [inline pkg/overlay/cache.go]: why did we change from empty string to randomID?

File: pkg/overlay/cache_test.go [hunk 3]
```diff
@@ -97,8 +99,10 @@ func bootstrapTestNetwork(t *testing.T, ip, port string) ([]dht.DHT, pb.Node) {
 
 		p++
 		dhts = append(dhts, dht)
-		err = dht.ListenAndServe()
-		assert.NoError(t, err)
+		go func() {
```
Comment (jenlij) [inline pkg/overlay/cache_test.go]: why do we want these in goroutines?

File: pkg/overlay/cache.go (hunk 2 above)
Comment (wthorp) [reply pkg/overlay/cache.go]: I don't understand any scenario where empty string is ideal.  Am I missing something?

Comment (wthorp) [reply pkg/overlay/cache.go]: Reading Dennis's new comments he added which describe what this function is supposed to do, I do believe empty string is best.

Comment (wthorp) [reply pkg/overlay/cache.go]: Hrmm, I guess that I misunderstood the DHT.GetNodes interface.  Reverting to empty string.

File: pkg/overlay/cache_test.go (hunk 1 above)
Comment (egonelbre) [reply pkg/overlay/cache_test.go]: This is would mean that you never shutdown the server.

**Pull request merged**